libraries/tools/kotlin-symbol-processing-api/src/org/jetbrains/kotlin/ksp/utils.kt

```diff
@@ -31,7 +31,7 @@ fun KSClassDeclaration.getDeclaredProperties(): List<KSPropertyDeclaration> {
  * Check whether this is a local declaration, or namely, declared in a function.
  */
 fun KSDeclaration.isLocal(): Boolean {
-    return this.parentDeclaration != null && this.parentDeclaration !is KSClassDeclaration
+    return (this.parentDeclaration != null && this.parentDeclaration !is KSClassDeclaration) || this.parentDeclaration?.isLocal() == true
```

Comment:
This conflicts with the definition in the [doc](https://kotlinlang.org/docs/reference/functions.html#function-scope).